CriteoPublisherSdk/Tests/UnitTests/CR_CacheManagerTests.m

```diff
@@ -123,6 +124,7 @@ - (void)testSetBidForInterstitial {
   OCMStub(deviceInfo.screenSize).andReturn(CGSizeMake(50, 320));
   [self.cacheManager setBid:self.testBid];
   XCTAssertTrue([[self.cacheManager getBidForAdUnit:self.adUnit] isEqual:self.testBid]);
+  Criteo.sharedCriteo.dependencyProvider.deviceInfo = _originDeviceInfo;
```

Comment:
Each method that are calling `mockDeviceInfo` should then restore the `dependencyProvider.deviceInfo`. What about putting this restoration in a `tearDown` method so that tests don't need to clean up internal states ?

Also, as the mock is a partial mock, it is fine to always have it. So why not setting the partial mock within the `setUp` method directly ? So all new tests could easily stub/verify the behavior:

```
@property originDeviceInfo;
@property spyDeviceInfo;

- setUp {
 _originDeviceInfo = dependencyProvider.deviceInfo;
 _spyDeviceInfo = OCMPartialMock(_originDeviceInfo);
 dependencyProvider.deviceInfo = _spyDeviceInfo;
}

- tearDown {
  dependencyProvider.deviceInfo = _originDeviceInfo;
}

- testFoo {
  OCMStub(self.spyDeviceInfo.screenSize).andReturn(...);
  ...
}
```